New Resource/Data Source: `azurerm_App_Service_Certificate_Order` (#4454)
terraform-providers/terraform-provider-azurerm: website/docs/d/app_service_certificate_order.html.markdown

sidebar_current: "docs-azurerm-datasource-app-service-x"
description: |-
  Gets information about an existing App Service Certificate Order.
---

# Data Source: azurerm_app_service_certificate_order

Use this data source to access information about an existing App Service Certificate Order.

## Example Usage

```hcl
data "azurerm_app_service_certificate_order" "test" {
    name                = "example-cert-order"
    resource_group_name = "example-resources"
}

output "certificate_order_id" {
    value = "${data.azurerm_app_service_certificate_order.test.id}
}
```

## Argument Reference

* `name` - (Required) The name of the App Service.

* `resource_group_name` - (Required) The Name of the Resource Group where the App Service exists.

## Attributes Reference

* `id` - The ID of the App Service.

* `location` - The Azure location where the App Service exists.

* `auto_renew` - true if the certificate should be automatically renewed when it expires; otherwise, false.

* `certificates` - State of the Key Vault secret. A `certificates` block as defined below.

* `csr` - Last CSR that was created for this order.

> * `distinguished_name` - Certificate distinguished name.

Reviewer: As tedious as it it we should expand on the descriptions here
```suggestion
* `distinguished_name` - The Distinguished Name for the App Service Certificate Order
```

* `key_size` - Certificate key size.

* `product_type` - Certificate product type.

* `validity_in_years` - Duration in years (must be between 1 and 3).

* `domain_verification_token` - Domain verification token.

* `status` - Current order status.

* `serial_number` - Current serial number of the certificate.

* `last_certificate_issuance_time` - Certificate last issuance time.

* `next_auto_renewal_time_stamp` - Time stamp when the certificate would be auto renewed next

* `expiration_time` - Certificate expiration time.

> * `is_private_key_external` - true if private key is external; otherwise, false.

Reviewer: We should clean up the language here
```suggestion
* `is_private_key_external` - Whether the private key is external or not. 
```
